src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig

```diff
@@ -235,6 +235,44 @@
         }
 
         {% if excluded_ajax_paths is defined %}
+
+            if (window.fetch) {
+                var oldFetch = window.fetch;
+                window.fetch = function () {
+                    var promise = oldFetch.apply(null, arguments);
+                    if (!arguments[0].match(new RegExp({{ excluded_ajax_paths|json_encode|raw }}))) {
+                        var method = 'GET';
+                        if (arguments[1] && arguments[1].method !== undefined) {
+                            method = arguments[1].method;
+                        }
+
+                        var stackElement = {
+                            loading: true,
+                            error: false,
+                            url: arguments[0],
+                            method: method,
+                            start: new Date()
+                        };
+
+                        requestStack.push(stackElement);
+                        promise.then(function (r) {
+                            stackElement.duration = new Date() - stackElement.start;
+                            stackElement.loading = false;
+                            stackElement.error = r.status < 200 || r.status >= 400;
+                            stackElement.statusCode = r.status;
+                            stackElement.profile = r.headers.get('x-debug-token');
+                            stackElement.profilerUrl = r.headers.get('x-debug-token-link');
+                            Sfjs.renderAjaxRequests();
+                        }).catch(function(err) {
+                            stackElement.loading = false;
+                            stackElement.error = true;
```

Comment:
what about adding a `type` so we could make distinction between `ajax (xhr)` and `fetch` in the toolbar. As they do not behave the same way, it may be useful, WDYT?